torchgeo/datasets/oscd.py

```diff
@@ -432,17 +472,19 @@ def setup(self, stage: Optional[str] = None) -> None:
         def n_random_crop(sample: Dict[str, Any]) -> Dict[str, Any]:
             images, masks = [], []
             for i in range(self.num_patches_per_tile):
-                mask = repeat(sample["mask"], "h w -> t h w", t=2)
+                mask = repeat(sample["mask"], "h w -> t h w", t=2).float()
                 image, mask = self.rcrop(sample["image"], mask)
                 mask = mask.squeeze()[0]
                 images.append(image.squeeze())
-                masks.append(mask)
+                masks.append(mask.long())
             sample["image"] = torch.stack(images)
             sample["mask"] = torch.stack(masks)
             return sample
 
         train_transforms = Compose([self.preprocess, n_random_crop])
-        test_transforms = Compose([self.preprocess])
+        # for testing and validation we pad all inputs to a fixed size to avoid issues
+        # with the upsampling paths in encoder-decoder architectures
+        test_transforms = Compose([self.preprocess, self.pad_to(1024 + 256)])
```

Comment:
Your train set returns patches but the test and val set don't. How are you planning to feed this data to a model for eval even though it hasn't seen patches that large during training?

Reply:
@isaaccorley for fully convolutional networks this shouldn't be a problem, right? i.e. there is nothing in a U-Net that depends on the size of the image

Reply:
That's correct. Now that I look at the image sizes I don't think this is a problem since they are generally around 512x512. For some reason I thought the image sizes varied more but it looks like only a few images are above 1000x1000.

I guess my only concern now is we are allowing a user to defined crop size but then hardcoding the test set padding to 1280x1280?


Here are the image sizes for clarity:

```python
# Train set
(785, 799)
(525, 471)
(772, 902)
(1070, 1180)
(360, 395)
(461, 517)
(788, 1015)
(540, 695)
(557, 858)
(582, 522)
(390, 408)
(718, 776)
(563, 339)
(688, 639)

# Test Set 
(469, 433)
(544, 730)
(634, 774)
(716, 824)
(558, 545)
(451, 426)
(385, 241)
(426, 353)
(688, 639)
(476, 458)
```

Reply:
What is your concern above? I picked 1280 because it is > all test set sizes and evenly divisible by 2 enough times to satisfy a U-Net.

Reply:
No concern then. Just interested to see how we handle that in the trainer.